globals/matchmaking_utils.go

```diff
@@ -140,6 +142,33 @@ func RemoveClientFromAllSessions(client *nex.Client) {
 	}
 }
 
+// CreateSessionByMatchmakeSession creates a gathering from a MatchmakeSession
+func CreateSessionByMatchmakeSession(matchmakeSession *match_making_types.MatchmakeSession, searchMatchmakeSession *match_making_types.MatchmakeSession, hostPID uint32) (*CommonMatchmakeSession, error, uint32) {
+	sessionIndex := GetAvailableGatheringID()
+	// This should in theory be impossible, as there aren't enough PIDs creating sessions to fill the uint32 limit.
+	// If we ever get here, we must be not deleting sessions properly
+	if sessionIndex == 0 {
+		return &CommonMatchmakeSession{}, fmt.Errorf("No gatherings available!"), nex.Errors.RendezVous.LimitExceeded
+	}
+
+	session := CommonMatchmakeSession{
+		SearchMatchmakeSession: searchMatchmakeSession,
+		GameMatchmakeSession:   matchmakeSession,
+	}
+
+	Sessions[sessionIndex] = &session
+	Sessions[sessionIndex].GameMatchmakeSession.Gathering.ID = sessionIndex
+	Sessions[sessionIndex].GameMatchmakeSession.Gathering.OwnerPID = hostPID
+	Sessions[sessionIndex].GameMatchmakeSession.Gathering.HostPID = hostPID
+
+	Sessions[sessionIndex].GameMatchmakeSession.StartedTime = nex.NewDateTime(0)
+	Sessions[sessionIndex].GameMatchmakeSession.StartedTime.UTC()
+	Sessions[sessionIndex].GameMatchmakeSession.SessionKey = make([]byte, 32)
+    rand.Read(Sessions[sessionIndex].GameMatchmakeSession.SessionKey)
+
```

Comment:
Possible micro-optimization here as well. Instead of doing many index lookups and then modifying the data, we could instead just set all the data on the `session` struct right away and only assign it to `Sessions` later. Unsure how much of a difference that would make but logically it should be faster?